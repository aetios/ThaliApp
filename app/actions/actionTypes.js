export const NAVIGATE = 'NAVIGATE';
export const BACK = 'BACK';
export const OPENDRAWER = 'OPENDRAWER';
export const CALENDARRETREIVED = 'CALENDARRETREIVED';
export const CALENDARERROR = 'CALENDARERROR';
export const RESETLOGINSTATE = 'RESETLOGINSTATE';
export const WELCOME = 'WELCOME';
