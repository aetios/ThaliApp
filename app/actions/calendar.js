/**
 * Created by aucke on 3/8/17.
 */
