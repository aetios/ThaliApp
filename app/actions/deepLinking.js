export const DEEPLINK = 'DEEPLINKING_DEEPLINK';

export function deepLink(url, stayInApp = true) {
  return {
    type: DEEPLINK,
    payload: { url, stayInApp },
  };
}
