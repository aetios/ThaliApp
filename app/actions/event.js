export const EVENT = 'EVENT_EVENT';
export const FETCHING = 'EVENT_FETCHING';
export const SUCCESS = 'EVENT_SUCCESS';
export const FAILURE = 'EVENT_FAILURE';

export function event(pk, token) {
  return {
    type: EVENT,
    payload: { pk, token },
  };
}

export function success(eventData, eventRegistrations) {
  return {
    type: SUCCESS,
    payload: { eventData, eventRegistrations },
  };
}

export function fetching() {
  return {
    type: FETCHING,
  };
}

export function failure() {
  return {
    type: FAILURE,
  };
}
