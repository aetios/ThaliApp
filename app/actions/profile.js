export const PROFILE = 'PROFILE_PROFILE';
export const FETCHING = 'PROFILE_FETCHING';
export const SUCCESS = 'PROFILE_SUCCESS';
export const FAILURE = 'PROFILE_FAILURE';

export function profile(member = 'me') {
  return {
    type: PROFILE,
    payload: { member },
  };
}

export function fetching() {
  return {
    type: FETCHING,
  };
}

export function success(profileData) {
  return {
    type: SUCCESS,
    payload: { profileData },
  };
}

export function failure() {
  return {
    type: FAILURE,
  };
}
