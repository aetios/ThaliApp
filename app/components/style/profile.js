import { colors, create } from '../../style';
import { APPBAR_HEIGHT } from './navigator';

export const HEADER_MIN_HEIGHT = APPBAR_HEIGHT;
export const HEADER_MAX_HEIGHT = 200;
export const HEADER_SCROLL_DISTANCE = HEADER_MAX_HEIGHT - HEADER_MIN_HEIGHT;

const styles = create({
  container: {
    backgroundColor: colors.background,
    flex: 1,
  },
  card: {
    backgroundColor: colors.white,
    marginLeft: 8,
    marginRight: 8,
    marginTop: 10,
    marginBottom: 10,
    ios: {
      borderRadius: 4,
      borderColor: colors.lightGray,
      borderStyle: 'solid',
      borderWidth: 0.5,
    },
    android: {
      elevation: 2,
      borderRadius: 2,
    },
  },
  profileText: {
    ios: {
      borderRadius: 4,
    },
  },
  item: {
    padding: 16,
  },
  borderTop: {
    borderTopColor: colors.dividerGrey,
    borderTopWidth: 1,
  },
  description: {
    fontSize: 14,
    lineHeight: 24,
    color: colors.black,
    android: {
      fontFamily: 'sans-serif-medium',
    },
    ios: {
      fontFamily: 'System',
      fontWeight: '600',
    },
  },
  data: {
    fontSize: 14,
    lineHeight: 24,
    color: colors.gray,
    android: {
      fontFamily: 'sans-serif-medium',
    },
    ios: {
      fontFamily: 'System',
      fontWeight: '600',
    },
  },
  url: {
    textDecorationLine: 'underline',
  },
  sectionHeader: {
    backgroundColor: colors.background,
    fontSize: 14,
    color: colors.textColour,
    marginLeft: 18,
    android: {
      fontFamily: 'sans-serif-medium',
    },
    ios: {
      fontFamily: 'System',
      fontWeight: '600',
    },
  },
  marginTop: {
    marginTop: 10,
  },
  italics: {
    fontStyle: 'italic',
  },
  header: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    backgroundColor: colors.magenta,
    overflow: 'hidden',
    elevation: 4,
  },
  appBar: {
    backgroundColor: colors.transparent,
    flex: 1,
    flexDirection: 'row',
    alignItems: 'flex-start',
    android: {
      justifyContent: 'flex-start',
      flexWrap: 'wrap',
    },
    ios: {
      borderStyle: 'solid',
      borderBottomColor: colors.darkMagenta,
      borderBottomWidth: 1,
      justifyContent: 'space-between',
    },
  },
  content: {
    marginTop: HEADER_MAX_HEIGHT,
  },
  icon: {
    fontSize: 24,
    marginTop: (HEADER_MIN_HEIGHT - 24) / 2,
    color: colors.white,
    android: {
      marginLeft: 16,
    },
    ios: {
      marginLeft: 10,
    },
  },
  title: {
    color: colors.white,
    position: 'absolute',
    android: {
      fontFamily: 'sans-serif-medium',
    },
    ios: {
      fontFamily: 'System',
      fontWeight: '600',
    },
  },
  backgroundImage: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    width: null,
    height: HEADER_MAX_HEIGHT,
    resizeMode: 'cover',
  },
  overlayGradient: {
    position: 'absolute',
    top: '50%',
    bottom: 0,
    left: 0,
    right: 0,
    opacity: 0.3,
  },
  errorText: {
    marginTop: HEADER_MIN_HEIGHT,
  },
});

export default styles;
