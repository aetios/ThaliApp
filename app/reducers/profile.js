import * as profileActions from '../actions/profile';

import { defaultProfileImage } from '../utils/url';

const initialState = {
  profile: {
    pk: -1,
    display_name: '',
    photo: defaultProfileImage,
    avatar: {
      full: defaultProfileImage,
      large: defaultProfileImage,
      medium: defaultProfileImage,
      small: defaultProfileImage,
    },
    profile_description: '',
    birthday: '',
    starting_year: -1,
    programme: '',
    website: '',
    membership_type: '',
    achievements: [],
    societies: [],
  },
  success: false,
  hasLoaded: false,
};

export default function profile(state = initialState, action = {}) {
  switch (action.type) {
    case profileActions.FETCHING:
      return {
        ...state,
        hasLoaded: false,
      };
    case profileActions.SUCCESS:
      return {
        ...state,
        profile: action.payload.profileData,
        success: true,
        hasLoaded: true,
      };
    case profileActions.FAILURE:
      return {
        ...state,
        success: false,
        hasLoaded: true,
      };
    default:
      return state;
  }
}
