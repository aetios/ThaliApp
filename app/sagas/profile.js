import { call, put, takeEvery } from 'redux-saga/effects';

import { apiRequest } from '../utils/url';
import * as profileActions from '../actions/profile';
import * as navActions from '../actions/navigation';

const profile = function* profile(action) {
  const { token, member } = action.payload;

  yield put(profileActions.fetching());
  yield put(navActions.navigate('profile'));

  const data = {
    method: 'GET',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Token ${token}`,
    },
  };

  try {
    const profileData = yield call(apiRequest, `members/${member}`, data);
    yield put(profileActions.success(profileData));
  } catch (error) {
    yield put(profileActions.failure());
  }
};

const profileSaga = function* eventSaga() {
  yield takeEvery(profileActions.PROFILE, profile);
};

export default profileSaga;
