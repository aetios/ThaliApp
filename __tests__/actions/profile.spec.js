import * as actions from '../../app/actions/profile';

describe('navigation actions', () => {
  it('should expose the navigation actions', () => {
    expect(actions.PROFILE).toEqual('PROFILE_PROFILE');
    expect(actions.FETCHING).toEqual('PROFILE_FETCHING');
    expect(actions.SUCCESS).toEqual('PROFILE_SUCCESS');
    expect(actions.FAILURE).toEqual('PROFILE_FAILURE');
  });

  it('should create an action to load a profile', () => {
    expect(actions.profile('token')).toMatchSnapshot();
    expect(actions.profile('abc123', 1)).toMatchSnapshot();
  });

  it('should create an action for profile fetch', () => {
    expect(actions.fetching()).toMatchSnapshot();
  });

  it('should create an action for a successful profile load', () => {
    expect(actions.success('profile')).toMatchSnapshot();
  });

  it('should create an action for a failed profile load', () => {
    expect(actions.failure()).toMatchSnapshot();
  });
});