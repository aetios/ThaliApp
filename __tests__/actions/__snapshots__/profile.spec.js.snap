// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation actions should create an action for a failed profile load 1`] = `
Object {
  "type": "PROFILE_FAILURE",
}
`;

exports[`navigation actions should create an action for a successful profile load 1`] = `
Object {
  "payload": Object {
    "profileData": "profile",
  },
  "type": "PROFILE_SUCCESS",
}
`;

exports[`navigation actions should create an action for profile fetch 1`] = `
Object {
  "type": "PROFILE_FETCHING",
}
`;

exports[`navigation actions should create an action to load a profile 1`] = `
Object {
  "payload": Object {
    "member": "me",
    "token": "token",
  },
  "type": "PROFILE_PROFILE",
}
`;

exports[`navigation actions should create an action to load a profile 2`] = `
Object {
  "payload": Object {
    "member": 1,
    "token": "abc123",
  },
  "type": "PROFILE_PROFILE",
}
`;
